docs/integration-guides/ipc-integration.md

```diff
@@ -0,0 +1,288 @@
+
+Here's the corresponding message definitions for the AccountWeight request and response types:
+
+```
+/** Returns the voting weight for the given account */
+table AccountWeight {
+    /** A nano_ address */
+    account: string (required);
+}
+
+/** Response to AccountWeight */
+table AccountWeightResponse {
+    /** Voting weight as a decimal number*/
+    voting_weight: string (required);
+}
+```
+
+---
+
+### Parsing errors
+
+Any problems with the JSON request will be reported with error details:
+
+```json
+{
+    "message_type": "Error",
+    "message": {
+        "code": 1,
+        "message": "Invalid message format: 3: 2: error: required field is missing: account in AccountWeight"
+    }
+}
+```
+
+---
+
+## IPC Authorization
+
+!!! warning "Work in progress"
+	Permission settings is a work in progress, and their exact definition and defaults will be part of RPC 2.0 in a future node release.
+
+With IPC 2.0, the Nano node offers an authorization system.
+
+The configuration is done in `config-access.toml` by defining users and optional roles. Permissions are then assigned to these. The node only checks for permissions, never roles. This way, you can freely structure roles and users the way that suits your situation.
+
+There is also a default user with limited default permissions, currently only allowed to use the `AccountWeight` and `IsAlive` calls. This is used when no credentials are given. The permissions of the default user can also be changed in the configuration file.
+
+Credentials:
+
+* IPC clients set the credentials in the message envelope 
+* HTTP(S) clients either use a message envelope or the HTTP Header `Nano-Api-Key`
+
+!!! tip "Layered security highly recommended"
+	While permissions enable node operators to pick what functionality to expose to which users, it is still highly recommended that layered security is used. For instance, a wallet backend should expose only required functionality to clients. The backend can then communicate with the node with credentials for additional security.
+
+### Call example
+
+```bash
+curl --header "Nano-Api-Key:mywalletuser" --insecure -d \
+   '{ "account": "nano_3t6k35gi95xu6tergt6p69ck76ogmitsa8mnijtpxm9fkcm736xtoncuohr3"}' \
+   https://www.example.com:7076/api/v2/AccountWeight
+```
+This uses HTTPS (which the node supports through a build option), and the `--insecure` is there because the node's certificate in this example is self-signed.
+
+Using an envelope instead of the `AccountWeight` endpoint:
+
+```json
+{ 
+   "credentials": "mywalletuser",
+   "message_type" : "AccountWeight", 
+   "message": 
+   {
+       "account": "nano_3t6k35gi95xu6tergt6p69ck76ogmitsa8mnijtpxm9fkcm736xtoncuohr3" 
+   }
+} 
+```
+
+`POST` the above to https://www.example.com:7076/api/v2
+
+### Configuration examples
+
+For testing IPC without caring about permissions, this gives access to everything:
+
+```toml
+[[user]]
+allow = "unrestricted"
+```
+
+A more elaborate sample:
+
+```toml
```

Comment:
Suggesting the second example is replaced with a bit smaller one. Maybe mention that the permissions are subject to change as part of the RPC 2.0 work?

```toml
[[role]]
id = "service_admin"
allow = "api_service_register, api_service_stop"

[[user]]
# User id's are typically randomly generated strings which
# matches the credentials in API requests.
id = "user-2bb818ee-6424-4750-8bdb-db23bab7bc57"

# Inherit all the permissions from these roles
roles = "service_admin"

# Add additional permissions for this specific user
allow = "wallet_seed_change, api_topic_confirmation"

# A list of specific permissions can be denied as well
deny = "api_account_weight"

[[user]]
id = "history-viewer-e3cf8a09-bd74-4ef2-9b84-e14f3db2bb4b"

# Add specific permission for this user
allow = "api_account_info, api_account_history"

# Do not inherit any default permissions. This is useful
# for making users with a explicit set of minimum permissions.
# The default user can also be set to bare. That way, a node can be
# exposed with a limited set of default permissions.
bare = true

```

Reply:
Added simpler perms example and extra WIP warning